packages/expo-sqlite/src/__tests__/Storage-test.ios.ts

```diff
@@ -60,6 +60,22 @@ describe('SQLiteStorage asynchronous', () => {
     const keys = await storage.getAllKeysAsync();
     expect(keys).toHaveLength(0);
   });
+
+  it('should get length', async () => {
+    await storage.setItemAsync('key1', 'value1');
+    await storage.setItemAsync('key2', 'value2');
+    const length = await storage.getLengthAsync();
+    expect(length).toBe(2);
+  });
+
+  it('should get key by index', async () => {
+    await storage.setItemAsync('key', 'value');
+    expect(await storage.getKeyByIndexAsync(0)).toBe('key');
+
+    // Invalid index should normalize to 0
+    expect(await storage.getKeyByIndexAsync(-1)).toBe('key');
+    expect(await storage.getKeyByIndexAsync(Number.MAX_VALUE)).toBe('key');
```

Comment:
I would expand this test with:
```ts
// Add a second item so the tests can distinguish between the first and last items
await storage.setItemAsync('key2', 'value2');
```
```suggestion
    expect(await storage.getKeyByIndexAsync(-1)).toBe(null);
    expect(await storage.getKeyByIndexAsync(0.5)).toBe('key');
    expect(await storage.getKeyByIndexAsync(true)).toBe('key2');
    expect(await storage.getKeyByIndexAsync(false)).toBe('key');
    expect(await storage.getKeyByIndexAsync({ valueOf: () => 1 })).toBe('key2');

    expect(await storage.getKeyByIndexAsync(Number.MAX_SAFE_INTEGER)).toBe(null);
    expect(await storage.getKeyByIndexAsync(Number.MAX_SAFE_INTEGER + 1)).toBe('key');
    expect(await storage.getKeyByIndexAsync(Number.MAX_VALUE)).toBe(null);
    expect(await storage.getKeyByIndexAsync(NaN)).toBe('key');
    expect(await storage.getKeyByIndexAsync(Infinity)).toBe('key');
    expect(await storage.getKeyByIndexAsync(-Infinity)).toBe('key');
```